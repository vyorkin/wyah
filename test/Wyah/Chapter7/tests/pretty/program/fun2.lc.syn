Program [Decl "ap" (ELam (Var "f") (ELam (Var "x") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x")))))),Decl "on" (ELam (Var "g") (ELam (Var "f") (ELam (Var "x") (ELam (Var "y") (EApp (EApp (EVar (Var "g")) (EApp (EVar (Var "f")) (EVar (Var "x")))) (EApp (EVar (Var "f")) (EVar (Var "y")))))))),Decl "twice" (ELam (Var "f") (ELam (Var "x") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x"))))))]
