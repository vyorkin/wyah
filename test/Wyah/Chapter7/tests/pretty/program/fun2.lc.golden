let ap = \x -> \f -> f (f x);
let on = \f -> \g -> \x -> \y -> (g (f x)) (f y);
let twice = \x -> \f -> f (f x);